lib/services/credentials/credential.service.js

```diff
@@ -45,10 +45,24 @@ s.insertCredential = function (id, type, credentialDetails) {
     throw new Error(`Invalid credential type: ${type}`); // TODO: replace with validation error
   }
 
-  return this.getCredential(id, type) // check if credential already exists
+  return credentialDao.getAllCredentials(id) // check if credential already exists
     .then((cred) => {
-      if (cred && cred.isActive) {
-        throw new Error('Credential already exists and is active'); // TODO: replace with validation error
+      if (Array.isArray(cred)) {
+        cred.forEach((c) => {
+          if (c && typeof c.isActive !== 'boolean') {
```

Comment:
Is this a condition that can even happen? I'm usually in for defensive programming but in this case I really do not see this happening. 🤔